kernel/kernel.go

```diff
@@ -27,13 +27,13 @@ import (
 	"regexp"
 
 	"gopkg.in/yaml.v2"
-
-	"github.com/snapcore/snapd/gadget/edition"
 )
 
 type Asset struct {
-	Edition edition.Number `yaml:"edition,omitempty"`
-	Content []string       `yaml:"content,omitempty"`
+	// TODO: we may make this an (optional) map at some point in
+	//       the future to select what things should be updated.
```

Comment:
Does it mean that unmarshalling will have to deal with ambiguity and try both bool and map?

Reply:
Yes, that was the goal, having a custom unmarshaler at some future point (if we go into this direction/this use-case comes up).